### PR TITLE
Replace JitIntrinsicAttribute with IntrinsicAttribute

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/Common/src/CoreLib/System/Numerics/Vector.tt [hunk 1]
```diff
@@ -66,15 +66,27 @@ namespace System.Numerics
         /// <summary>
         /// Returns a vector containing all zeroes.
         /// </summary>
-        [JitIntrinsic]
-        public static Vector<T> Zero { get { return zero; } }
-        private static readonly Vector<T> zero = new Vector<T>(GetZeroValue());
+        public static Vector<T> Zero
+        { 
+            [Intrinsic]
```
Comment (jkotas) [inline src/Common/src/CoreLib/System/Numerics/Vector.tt]: The Intrinsic attribute has to be on the method bodies, not on the properties.

File: src/Common/src/CoreLib/System/Numerics/Vector.cs [hunk 2]
```diff
@@ -4134,12 +4146,12 @@ internal static unsafe Vector<T> Max(Vector<T> left, Vector<T> right)
...
         }
 
-        [JitIntrinsic]
+        [Intrinsic]
         internal static T DotProduct(Vector<T> left, Vector<T> right)
         {
             if (Vector.IsHardwareAccelerated)
             {
-                T product = GetZeroValue();
+                T product = default;
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/Numerics/Vector.cs]: GetZeroValue was throwing an exception for unsupported types. Did that provide any meaningful benefit here, i.e. without it are we losing some protection for something?

File: src/Common/src/CoreLib/System/Numerics/Vector.tt [hunk 3]
```diff
@@ -66,15 +66,27 @@ namespace System.Numerics
...
+            get
+            {
+                return zero;
+            } 
+        }
+        private static readonly Vector<T> zero = new Vector<T>();
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/Numerics/Vector.tt]: Nit: s_zero

File: src/Common/src/CoreLib/System/Numerics/Vector.cs [hunk 4]
```diff
@@ -61,15 +61,27 @@ public static int Count
...
-        private static readonly Vector<T> zero = new Vector<T>(GetZeroValue());
+        public static Vector<T> Zero
+        {
+            [Intrinsic]
+            get
+            {
+                return zero;
+            }
+        }
+        private static readonly Vector<T> zero = new Vector<T>();
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/Numerics/Vector.cs]: Nit: s_zero, s_one, etc., here and elsewhere this occurs

**Comment by akoeplinger** [comment]
@marek-safar I see we're checking for this attribute here: https://github.com/mono/mono/blob/c0dd0232af958add2754bd59f91e2a75f74c8fef/mono/mini/simd-intrinsics.c#L2049-L2050

Is it going to pop up on your next corefx bump automatically or should I file a tracking issue?

**Comment by marek-safar** [comment]
@akoeplinger should be a build error for us (IntristicAttribute is still internal) so we notice on next bump

File: src/Common/src/CoreLib/System/Numerics/Vector.cs (hunk 2 above)
Comment (jkotas) [reply src/Common/src/CoreLib/System/Numerics/Vector.cs]: The exception for unsupported types is still going to be thrown here by `ScalarAdd` a few lines below.

There should be no changes in exceptions thrown from this delta (here and elsewhere). The exception may just be thrown at slightly different place.

Comment (stephentoub) [reply src/Common/src/CoreLib/System/Numerics/Vector.cs]: Cool

**Comment by jkotas** [comment]
@marek-safar @akoeplinger Thanks for catching the Mono dependency on the attribute name!

**Pull request merged**